Convert React components to functions

ilong4rennes opened the pull request:
## What:
- Converted React class components to function components for 48 out of the 49 files are listed in issue #571. Updates include replacing class-based lifecycle methods with React hooks and ensuring the components maintain the same behavior as before.
- The remaining file, admin/src/Models/CollectionNavigation.js, contains a CollectionNavigation class that serves as a base class for multiple other components. To maintain consistency and avoid introducing incomplete changes, we opted to leave this class unchanged for now. After investigating related dependencies, a new issue has been created to address this component separately.

## Why:
Modernizing the codebase to align with current React best practices. Function components are more concise, easier to read, and enable the use of modern React features like hooks. 

## Testing:
- ESLint passes during `make test`
- Manual Testing UI: Verified the UI functionality of all updated components. Screenshots of the results are included in the related PRs, available at [this link](https://github.com/ilong4rennes/makeradmin/pulls?q=is%3Apr+is%3Aclosed).

emanuelen5 commented:
The ruff test failed on main, which is why it failed on this PR as well. However, I've fixed it  now on main, so you can merge from main to get that passing.

emanuelen5 commented:
Thank you very much for the contributions!

There are a few regressions:
- [ ] The member spans page is broken: http://localhost:8009/membership/members/1/spans
    Before: 
![image](https://github.com/user-attachments/assets/612f71d3-11e3-4d1d-acf2-9ea425687212)
    Now: The page is blank with the following errors in the logs ![image](https://github.com/user-attachments/assets/702a9354-a904-4de2-9d97-deffc3ba0055)
- [ ] The span info page, http://localhost:8009/membership/spans/1, causes infinite network requests, DoS:ing my computer.
- [ ] The span info page, http://localhost:8009/membership/spans/1, does not render correctly - the fields are not filled in.
    Before: 
![image](https://github.com/user-attachments/assets/7ad74a50-c4a0-441b-9892-e214e0cfa962)
    Now: 
![image](https://github.com/user-attachments/assets/7bf7eb8f-c5f7-4a70-b902-0b6be99e9787)
- [ ] The email log page is blank: http://localhost:8009/messages/1
    Before: 
![image](https://github.com/user-attachments/assets/1427cf3d-ef8b-49a1-9bfc-74fcc009a593)
    Now: 
![image](https://github.com/user-attachments/assets/774b3b3c-e2de-44f4-8310-606be772207c)
- [ ] The order page http://localhost:8009/sales/order/1 causes infinite network requests.
- [ ] The order page http://localhost:8009/sales/order/1 does not show the correct member who performed the purchase.
    Before: 
![image](https://github.com/user-attachments/assets/16d669ce-87b5-4e91-ae46-add8f731bd22)
    Now: 
![image](https://github.com/user-attachments/assets/a2be89bd-c56b-4e22-ac60-98aa2725d616)
- [ ] The gift-card page, http://localhost:8009/sales/gift-card/1, causes infinite network requests.
- [ ] The gift-card page does not render correctly.
    Before: 
![image](https://github.com/user-attachments/assets/7bb3251e-e05c-406b-a1a3-002b0522ed59)
    Now: 
![image](https://github.com/user-attachments/assets/b29acc3a-80fd-4def-bced-34db47b84451)

HalfVoxel commented:
What a great contribution <3! Amazing work!

emanuelen5 reviewed (commented):
Looks really good! I've reviewed about halfway. I had some comments.

emanuelen5 commented:
> Thank you very much for the contributions!
> 
> I haven't looked through all pages, but I noted that there are a few regressions:
> 
>     * [x]  The member spans page is broken: http://localhost:8009/membership/members/1/spans
>       Before:
>       ![image](https://private-user-images.githubusercontent.com/7572427/393543480-612f71d3-11e3-4d1d-acf2-9ea425687212.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3NzUxNjcsIm5iZiI6MTczMzc3NDg2NywicGF0aCI6Ii83NTcyNDI3LzM5MzU0MzQ4MC02MTJmNzFkMy0xMWUzLTRkMWQtYWNmMi05ZWE0MjU2ODcyMTIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDlUMjAwNzQ3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTk5NjVkNWFkMTYxYzY0NDM3YmY1MmZlMDNmMjA3NWU4YTZjMjE2NDgwY2ExMzI3MTY3ZGUyM2UwNzQyODYyNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.PwZBGC77OpMAG9LkXQoInEqNXsVvrS8U6raVCHE9RqE)
>       Now: The page is blank with the following errors in the logs ![image](https://private-user-images.githubusercontent.com/7572427/393542894-702a9354-a904-4de2-9d97-deffc3ba0055.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3NzUxNjcsIm5iZiI6MTczMzc3NDg2NywicGF0aCI6Ii83NTcyNDI3LzM5MzU0Mjg5NC03MDJhOTM1NC1hOTA0LTRkZTItOWQ5Ny1kZWZmYzNiYTAwNTUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDlUMjAwNzQ3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2U5MWIwZTRmOGI1ZGU3M2VjMTE3MzUzODhjNGYyZmE5YTlmMTQ2NmRiMzYxN2RmMzczMGE0ZDI5YTFiYmUyOCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.GvloMzREIiEHuy69c__LaSPkF4yJBp6X_9dAq_SB_T8)
> 
>     * [x]  The span info page, http://localhost:8009/membership/spans/1, causes infinite network requests, DoS:ing my computer.
> 
>     * [x]  The span info page, http://localhost:8009/membership/spans/1, does not render correctly - the fields are not filled in.
>       Before:
>       ![image](https://private-user-images.githubusercontent.com/7572427/393543548-7ad74a50-c4a0-441b-9892-e214e0cfa962.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3NzUxNjcsIm5iZiI6MTczMzc3NDg2NywicGF0aCI6Ii83NTcyNDI3LzM5MzU0MzU0OC03YWQ3NGE1MC1jNGEwLTQ0MWItOTg5Mi1lMjE0ZTBjZmE5NjIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDlUMjAwNzQ3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWMwMTQ5YTI0NDNiM2E2NjcyNDk0MDgxOTZlMDc5NTEyNzMzN2Y0ZjhkOGYzOGY2NTE2NGJiZjZlZTEyMzZkMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.73BspnoWEVALK5sZf_7FkRnFB6P5hAtyXIqjLJzRBNI)
>       Now:
>       ![image](https://private-user-images.githubusercontent.com/7572427/393543617-7bf7eb8f-c5f7-4a70-b902-0b6be99e9787.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3NzUxNjcsIm5iZiI6MTczMzc3NDg2NywicGF0aCI6Ii83NTcyNDI3LzM5MzU0MzYxNy03YmY3ZWI4Zi1jNWY3LTRhNzAtYjkwMi0wYjZiZTk5ZTk3ODcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDlUMjAwNzQ3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDgxOGRlNDRiNDYxNGY3YmJkYmZiM2E0YzRiYTFjYmRkMDFhYzQ2ZGI0N2UxZjA4ODgxNDcyMjE5OTc2YjRlZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.bXfFJ4xCCJwkqdCK1Klliv5K-phRFBOn2zZGmf-F-Z8)
> 
>     * [x]  The email log page is blank: http://localhost:8009/messages/1
>       Before:
>       ![image](https://private-user-images.githubusercontent.com/7572427/393543427-1427cf3d-ef8b-49a1-9bfc-74fcc009a593.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3NzUxNjcsIm5iZiI6MTczMzc3NDg2NywicGF0aCI6Ii83NTcyNDI3LzM5MzU0MzQyNy0xNDI3Y2YzZC1lZjhiLTQ5YTEtOWJmYy03NGZjYzAwOWE1OTMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDlUMjAwNzQ3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTlkMjlhNjVkYWYwNzY2MGU1ZWIwYmQ5YWMyMjA0MGZiODNhMWY0YjNmNTAyYmEwNjcwYzc0NTUwNDk1YzIwZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.RHb_8zXrojR6U1yYnVtF1m_aazmNmUC5t29NqALXsng)
>       Now:
>       ![image](https://private-user-images.githubusercontent.com/7572427/393543677-774b3b3c-e2de-44f4-8310-606be772207c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3NzUxNjcsIm5iZiI6MTczMzc3NDg2NywicGF0aCI6Ii83NTcyNDI3LzM5MzU0MzY3Ny03NzRiM2IzYy1lMmRlLTQ0ZjQtODMxMC02MDZiZTc3MjIwN2MucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDlUMjAwNzQ3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDY1NTc4YWUwN2M2ZjA3YWJiYjVkYWRjMzQ3ZWUzOWE3ODFiOTJjMGE1ZTc3M2UwNDE0YzE5MDkwZGM1NzFiOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.cI66bat1RkyczP6wnlfrUuRtxMcjH8Ua2MZoAcYZJa8)
> 
>     * [x]  The order page http://localhost:8009/sales/order/1 causes infinite network requests.
> 
>     * [x]  The order page http://localhost:8009/sales/order/1 does not show the correct member who performed the purchase.
>       Before:
>       ![image](https://private-user-images.githubusercontent.com/7572427/393543836-16d669ce-87b5-4e91-ae46-add8f731bd22.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3NzUxNjcsIm5iZiI6MTczMzc3NDg2NywicGF0aCI6Ii83NTcyNDI3LzM5MzU0MzgzNi0xNmQ2NjljZS04N2I1LTRlOTEtYWU0Ni1hZGQ4ZjczMWJkMjIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDlUMjAwNzQ3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NmUwMGZmY2UyZDhkNTQ0NTQwMzExNzg5YjMyNmI0Y2NlYzQ3MjMzYzI4MjZjZTA1NzE4NTQyNDU1ZjcyMjc4MCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.mObT6k2C6K-AZw2yQng7C-t-5Sf7pgbZ5WmgF1c_phs)
>       Now:
>       ![image](https://private-user-images.githubusercontent.com/7572427/393543790-a2be89bd-c56b-4e22-ac60-98aa2725d616.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3NzUxNjcsIm5iZiI6MTczMzc3NDg2NywicGF0aCI6Ii83NTcyNDI3LzM5MzU0Mzc5MC1hMmJlODliZC1jNTZiLTRlMjItYWM2MC05OGFhMjcyNWQ2MTYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDlUMjAwNzQ3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTdjNmJmYmI1M2NlMjAwYzk5NTYyNmUyYjczOThiMDdjNmM5MjRmMWY0ZTkzZjY1MmVkNzQyMDdhMmVkZDI2ZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.h_d2NluoHxaBUKgN4V-fA615wKdmdnE3p3rmHXYybBQ)
> 
>     * [x]  The gift-card page, http://localhost:8009/sales/gift-card/1, causes infinite network requests.
> 
>     * [x]  The gift-card page does not render correctly.
>       Before:
>       ![image](https://private-user-images.githubusercontent.com/7572427/393544044-7bb3251e-e05c-406b-a1a3-002b0522ed59.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3NzUxNjcsIm5iZiI6MTczMzc3NDg2NywicGF0aCI6Ii83NTcyNDI3LzM5MzU0NDA0NC03YmIzMjUxZS1lMDVjLTQwNmItYTFhMy0wMDJiMDUyMmVkNTkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDlUMjAwNzQ3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YmI4YTVlOTI1MWZhZTg5MzFmMWMwM2JkZGNlM2ZhMmE5MGMxZDc3ZThjYjI4NWNkNGQwYjBhYmU5NzM4NmQwZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.onj0WS0OdV0sNgP-S_drMMjuH4wnZqVfmI-LipDjRZA)
>       Now:
>       ![image](https://private-user-images.githubusercontent.com/7572427/393544010-b29acc3a-80fd-4def-bced-34db47b84451.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3NzUxNjcsIm5iZiI6MTczMzc3NDg2NywicGF0aCI6Ii83NTcyNDI3LzM5MzU0NDAxMC1iMjlhY2MzYS04MGZkLTRkZWYtYmNlZC0zNGRiNDdiODQ0NTEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDlUMjAwNzQ3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTlmMzkxMmRkMjkxYjdkMmNhNDYwMTQwOWE2NTUzMzE4MjFlN2IyZjFjMWUyNTU1MjE3Nzg1MGM5MjQ3N2FkYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.-sfxLOA-Hdr3dckbVb2Kf2rPHI7eFhKXyKcK-HBaXeE)

All of these regressions seem fixed now, thank you!

ilong4rennes commented:
Thank you so much for your detailed feedback, will address these issues later.

ilong4rennes commented:
Hi, sorry for the delay in response, I've updated the code according to your feedback. 
- deleted comments, unused hooks, removed unnecessary withRouter
- changed from e.preventDefault(); to return false;
- turned into a controlled component in RequestPasswordReset.jsx

I have not address these two comments:
1. rename it to DateTimeDisplay in [KeyEdit.jsx](https://github.com/makerspace/makeradmin/pull/581/files/d69424c14726e8b96ffc71a8fdc99f53abcc9873#diff-11fff720d3c7e59f52dcb0eb687bcb25939416b95d1e4af286fc0a9f55cd3f00), I will make a separate PR later
2. GroupBoxPermissions.jsx: I'll revisit coderabbit comment and update this later.

Thanks again!

emanuelen5 reviewed (approved):
There are a few things we should clean up, but we can do all of that in follow-up PRs. This seems to work, and is quite the improvement. Thanks a lot! ❤️
